e2e/support/test_support.go

```diff
@@ -1079,6 +1079,16 @@ func IntegrationTraitProfile(t *testing.T, ctx context.Context, ns string, name
 	}
 }
 
+func IntegrationTraitSpec(t *testing.T, ctx context.Context, ns string, name string) func() *v1.Traits {
```

Comment:
I'd remove this as well as it makes sense in unit test. In E2E we want to test the behavior instead.